Main/Controllers/UsersController.cs

```diff
@@ -31,5 +31,29 @@ public List<UserDTO> GetFriends()
       var users = _dbContext.ApplicationUser.Where(user => user.Id != userId);
       return users.Select(user => UserDTO.FromApplicationUser(user)).ToList();
     }
+
+    [Authorize(Roles = "Administrator")]
+    [HttpDelete("{userId}")]
+    public IActionResult DeleteUser(string userId)
+    {
+      var user = _dbContext.ApplicationUser
+        .Include(u => u.Collections)
+        .SingleOrDefault(u => u.Id == userId);
+
+      if (user == null)
+        return NotFound();
+
+      user.Collections.ToList().ForEach(c =>
```

Comment:
If you have an IEnumerable, it is better to use a simple forreach llop instead of forcing it to list and using linq